### DISCUSSION TITLE
Group Invoices By Project

### DISCUSSION THREAD

**Discussion by OkOmni** [kimai/kimai#3851] [General]
Do you know if there is a way to group invoices by project? I'm wondering if there's a plugin out there, free or paid, that can help with this. Or maybe it's possible to set up by default? Thanks for any help you can provide!

**Comment by labor4** [comment]
I asked your question more specifically to twig exports [in this discussion](https://github.com/kimai/kimai/discussions/3875)

**Comment by kevinpapst** [comment]
Invoice template: 

<img width="358" alt="Bildschirm­foto 2023-02-24 um 15 34 44" src="https://user-images.githubusercontent.com/533162/221204649-17d16d92-9e34-4604-a040-6faffbe7061d.png">

**Reply by OkOmni** [reply]
This would work in some cases, but we have a customer that has dozens of active projects. Because we ultimately need to see the customers full scope we can not segment the projects as their own customer.

as for the invoicing part, this works to a point that it would be great (ideal even) to have the one entry per time entry and grouped by project.

**Reply by labor4** [reply]
> as for the invoicing part, this works to a point that it would be great (ideal even) to have the one entry per time entry and grouped by project.

yes, a multidimensional array.
I'm trying to re-fill the array grouped by project in twig itself, but uhh... no luck so far.

**Reply by OkOmni** [reply]
yea, that was my take on it as well. I was hoping maybe it was an easy solution or that someone else might have done this for their own needs.

Thanks anyway~

**Comment by HeinzWuert** [comment]
Are you looking for invoices only or for an overlook based on customers.
Maybe this is going in the same direction: [Report for management regarding customers #3880](https://github.com/kimai/kimai/issues/3880)

**Comment by OkOmni** [comment]
I know there are reports which I can run individually within Kimai, but ideally I get a list of projects similar to the list of customers under invoices.

I need to be able to track what has been billed and what has not been billed.
